prtpy/partitioning/recursive_number_partitioning_sy_v2.py

```diff
@@ -0,0 +1,190 @@
+"""
+An implementation of full branch and bound search for number partitioning, with minmax objective
+(i.e. minimize largest sum). A recursive implementation.
+Based on "Search Strategies for Optimal Multi-Way Number Partitioning", Michael D. Moffitt, 2013
+https://www.ijcai.org/Proceedings/13/Papers/099.pdf
+
+note: As this is a full search algorithm, and the running time is exponential,
+it may not be used with large amount of items to partition.
+
+Author: nir son
+"""
+import doctest
+import math
+import random
+import time
+from numbers import Number
+from typing import List
+from prtpy import partition, Binner, BinnerKeepingContents, BinsArray, printbins
+from prtpy.partitioning.greedy import greedy
+import logging
+
+LOG_FORMAT = "%(levelname)s, time: %(asctime)s ,in: %(funcName)s , line: %(lineno)d: %(message)s "
+logging.basicConfig(format=LOG_FORMAT)
```

Comment:
Commands that affect the global logging behavior should not be in a module - they belong to the main program only.